File: backend/schemas/external_cocktails.py
from typing import List, Optional

from pydantic import BaseModel, Field


class ExternalCocktailItem(BaseModel):
    external_id: str
    name: str
    thumb_url: Optional[str] = None
    category: Optional[str] = None


class ExternalCocktailSearchResponse(BaseModel):
    items: List[ExternalCocktailItem] = Field(default_factory=list)
    degraded: bool = False
    message: Optional[str] = None

File: frontend/src/api/externalCocktails.ts
import api from './auth';

export type ExternalCocktailItem = {
  external_id: string;
  name: string;
  thumb_url?: string | null;
  category?: string | null;
};

export type ExternalCocktailSearchResponse = {
  items: ExternalCocktailItem[];
  degraded: boolean;
  message?: string | null;
};

export const externalCocktailsAPI = {
  searchByName: async (name: string): Promise<ExternalCocktailSearchResponse> => {
    const response = await api.get<ExternalCocktailSearchResponse>('/external/cocktails/search-by-name', {
      params: { name },
    });
    return response.data;
  },
};
